2020/day8/solution.py

```diff
@@ -0,0 +1,75 @@
+import re
+import typing
+from collections.abc import Iterable, Callable
+
+#Type Information
+Instruction = tuple[str, int]
+
+instruction_re = re.compile(r"(?P<instruction>\w{3}) (?P<value>[+-]\d+)")
+
+class Machine:
+
+    def _acc(self, value : int) -> int:
+        self.acc += value
+        return 1
+
+    INSTRUCTION_TABLE : dict[str, Callable[['Machine',int], int]] = {
+        'acc' : _acc,
+        'jmp' : lambda self, x: x,
+        'nop' : lambda self, x: 1,
+    }
+
+    def __init__(self, instructions : Iterable[Instruction]):
+        self.instructions = list(instructions)
+        self.reset()
+
+    def execute(self) -> bool:
+        self.reset()
+        executed_instructions = set()
+        while self.ip not in executed_instructions:
+            executed_instructions.add(self.ip)
+            if self.ip >= len(self.instructions):
+                return True
+            self.ip += self.execute_instruction(self.ip)
+        return False
+
+    def execute_instruction(self, ip : int) -> int:
```

Comment:
```suggestion
    def _execute_instruction(self, ip : int) -> int:
```
Prefixing with an underscore signals "private" for fields/methods external callers shouldn't touch (assuming folks should use a higher-level method like execute or repair here).

Reply:
thx